lib/BackgroundJobs/InitialContentImportJob.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Nextcloud - ContextChat
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Anupam Kumar <kyteinsky@gmail.com>
+ * @copyright Anupam Kumar 2024
+ */
+
+declare(strict_types=1);
+
+namespace OCA\ContextChat\BackgroundJobs;
+
+use OCA\ContextChat\Public\IContentProvider;
+use OCA\ContextChat\Service\ProviderConfigService;
+use OCP\App\IAppManager;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\QueuedJob;
+use OCP\IUserManager;
+use OCP\Server;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
+use Psr\Log\LoggerInterface;
+
+class InitialContentImportJob extends QueuedJob {
+	public function __construct(
+		private IAppManager $appManager,
+		private ProviderConfigService $configService,
+		private LoggerInterface $logger,
+		private IUserManager $userMan,
+		ITimeFactory $timeFactory,
+		private ?string $userId,
+	) {
+		parent::__construct($timeFactory);
+	}
+
+	/**
+	 * @param class-string<IContentProvider> $argument Provider class name
+	 * @return void
+	 */
+	protected function run($argument): void {
+		if (!is_string($argument)) {
+			return;
+		}
+
+		try {
+			/** @var IContentProvider */
+			$providerObj = Server::get($argument);
+		} catch (ContainerExceptionInterface | NotFoundExceptionInterface $e) {
+			$this->logger->warning('Could not run initial import for content provider', ['exception' => $e]);
+			return;
+		}
+
+		$user = is_null($this->userId) ? null : $this->userMan->get($this->userId);
```

Comment:
I think there's usually no user set in the cron.php environment.

Reply:
Also, just for good measure, PHPStorm always tells me `=== null` is faster than `is_null`, not sure if that's true :D